chronicle/release/releasers/github/summarizer_test.go

```diff
@@ -1188,3 +1191,275 @@ func toJson(t *testing.T, changes []change.Change) string {
+			config: Config{
+				ConsiderPRMergeCommits: true,
+			},
+			releaseFetcher: func(_, _, tag string) (*ghRelease, error) {
+				assert.Equal(t, "v0.2.0", tag)
+				return &ghRelease{
+					Tag:      tag,
+					Date:     testTime,
+					IsLatest: true,
+					IsDraft:  false,
+				}, nil
+			},
+			want: &changeScope{
+				Commits: gitLogRange(t, "test-fixtures/repos/v0.3.0-dev-repo", "v0.2.0", "", false),
+				Start: changePoint{
+					Ref: "v0.2.0",
+					Tag: &git.Tag{
+						Name:      "v0.2.0",
+						Timestamp: testTime,
+						Commit:    gitTagCommit(t, "test-fixtures/repos/v0.3.0-dev-repo", "v0.2.0"),
+					},
+					Inclusive: false,
+					Timestamp: &testTime,
+				},
+				End: changePoint{
+					Ref:       gitHeadCommit(t, "test-fixtures/repos/v0.3.0-dev-repo"),
+					Tag:       nil,
+					Inclusive: true,
+					Timestamp: nil,
+				},
+			},
+		},
+		{
+			name:        "first release (already has start tag)",
+			repoFixture: "test-fixtures/repos/v0.3.0-dev-repo",
+			sinceRef:    "v0.2.0",
+			untilRef:    "",
+			config: Config{
+				ConsiderPRMergeCommits: true,
+			},
+			releaseFetcher: func(_, _, _ string) (*ghRelease, error) {
+				return nil, nil
+			},
+			want: &changeScope{
+				Commits: gitLogRange(t, "test-fixtures/repos/v0.3.0-dev-repo", "v0.2.0", "", false),
+				Start: changePoint{
+					Ref: "v0.2.0",
+					Tag: &git.Tag{
+						Name:      "v0.2.0",
+						Timestamp: testTime,
+						Commit:    gitTagCommit(t, "test-fixtures/repos/v0.3.0-dev-repo", "v0.2.0"),
+					},
+					Inclusive: false,
+					Timestamp: nil, // this is the difference between this test and the previous one
+				},
+				End: changePoint{
+					Ref:       gitHeadCommit(t, "test-fixtures/repos/v0.3.0-dev-repo"),
+					Tag:       nil,
+					Inclusive: true,
+					Timestamp: nil,
+				},
+			},
+		},
+		{
+			name:        "first release (no tag found)",
+			repoFixture: "test-fixtures/repos/v0.1.0-dev-repo",
+			sinceRef:    "",
+			untilRef:    "",
+			config: Config{
+				ConsiderPRMergeCommits: true,
+			},
+			releaseFetcher: func(_, _, _ string) (*ghRelease, error) {
+				return nil, nil
+			},
+			want: &changeScope{
+				Commits: gitLogRange(t, "test-fixtures/repos/v0.1.0-dev-repo", "", "", false),
+				Start: changePoint{
+					Ref:       gitFirstCommit(t, "test-fixtures/repos/v0.1.0-dev-repo"),
+					Tag:       nil,
+					Inclusive: true, // this is the difference between this test and others
+					Timestamp: nil,
+				},
+				End: changePoint{
+					Ref:       gitHeadCommit(t, "test-fixtures/repos/v0.1.0-dev-repo"),
+					Tag:       nil,
+					Inclusive: true,
+					Timestamp: nil,
+				},
+			},
+		},
+	}
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			if tt.wantErr == nil {
+				tt.wantErr = assert.NoError
+			}
+
+			gitter, err := git.New(tt.repoFixture)
+			require.NoError(t, err)
+
+			// ensure that the times are sourced similarly... but the real values don't need to be under test
+			gitter = mockGitter{
```

Comment:
This reassignment and naming confused me a little bit. It looks like you make a real gitter, and a mock gitter, but the mock gitter has a pointer to the real gitter, and you can assign either the mock or real gitter to a variable called `gitter`. Maybe the comment here could be rephrased, or one of the `gitter` vars could get a different name?

Reply:
I'll update the comment, using the same name is intentional: any usage of the unmocked gitter would be invalid from that point forward.